CMakeLists.txt

```diff
@@ -65,198 +60,180 @@ include (GitRepo)
 set(xournalpp_CXX_FLAGS ${CMAKE_CXX_FLAGS})
 
 if (MSVC)
-	set(xournalpp_CXX_FLAGS ${xournalpp_CXX_FLAGS} -DNOMINMAX -DWIN32_LEAN_AND_MEAN)
-else ()
-    set(LIB_ENDING "so")
+    set(xournalpp_CXX_FLAGS ${xournalpp_CXX_FLAGS} -DNOMINMAX -DWIN32_LEAN_AND_MEAN)
 endif ()
 
 ## Libraries ##
-
-macro(add_includes_ldflags LDFLAGS INCLUDES)
-	set(xournalpp_LDFLAGS ${xournalpp_LDFLAGS} "${LDFLAGS}")
-    include_directories(SYSTEM ${INCLUDES})
-    # Note: System headers and external libraries are  intentionally
-    #       NOT added to the additional include paths
-endmacro (add_includes_ldflags LDFLAGS INCLUDES)
-
+##############################
+# c++17 support
+##############################
 find_package(CXX17 REQUIRED COMPONENTS optional)
 find_package(Filesystem REQUIRED COMPONENTS ghc Final Experimental)
 
-find_package(X11)
-if(X11_FOUND AND X11_Xi_FOUND)
-	add_definitions (-DX11_ENABLED)
-	add_includes_ldflags ("${X11_X11_LIB}" "${X11_INCLUDE_DIR}")
-	add_includes_ldflags ("${X11_Xi_LIB}" "${X11_Xi_INCLUDE_PATH}")
-endif()
-
-
-# libexec
 if (WIN32)
-	set(xournalpp_LDFLAGS ${xournalpp_LDFLAGS} "-mwindows")
-	# Need ImageMagick to generate icons on Windows
-	find_package(ImageMagick REQUIRED COMPONENTS convert)
-else ()
-	find_package(Backtrace REQUIRED)
-	add_includes_ldflags("${Backtrace_LIBRARIES}" "${Backtrace_INCLUDE_DIRS}")
+    # libexec
+    # Todo does not belong here
```

Comment:
Could you either fix this or describe what needs to be done? This way following up on your work is hard.